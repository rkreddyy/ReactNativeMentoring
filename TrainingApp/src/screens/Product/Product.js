import React from 'react';
import { View, Image, Text } from 'react-native';

import Styles from './styles';

const Product = props => {
    getPriceString = (item) => {
        return `$${item.price}`;
    }

    getOldPriceString = (item) => {
        return `$${item.oldPrice}`;
    }

    getDiscountString = (item) => {
        return `${item.discount}% Off`;
    }

    return (
        <View style={Styles.wrapper}>
            <View style={Styles.imageWrapper}>
                <Image source={props.item.source} style={Styles.image} />
            </View>
            <Text>{props.item.name}</Text>
            <View style={Styles.priceWrapper}>
                <Text style={Styles.price}>{getPriceString(props.item)}</Text>
                <Text style={Styles.oldPrice}>{getOldPriceString(props.item)}</Text>
                <Text style={Styles.discount}>{getDiscountString(props.item)}</Text>
            </View>
        </View>
    );
}

export default Product;
