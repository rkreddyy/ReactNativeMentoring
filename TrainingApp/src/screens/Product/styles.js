import {StyleSheet} from 'react-native';

import {BaseStyles} from '../../app.styles';

const ProductStyles = StyleSheet.create({
    wrapper: {
        marginRight: 'auto',
        flexBasis: 190,
        flexShrink: 2,
        marginBottom: BaseStyles.margin.xs,
        padding: BaseStyles.padding.m,

        borderRadius: 5,
        borderWidth: 1,
        borderColor: BaseStyles.colors.lightGrey,
    },
    imageWrapper: {
        alignItems: 'center',
        justifyContent: 'center',
        alignSelf: 'stretch',
        overflow: 'hidden',
        marginTop: BaseStyles.margin.s,
    },
    image: {
        height: 90,
        width: 40,
    },
    priceWrapper: {
        flexDirection: 'row',
    },
    price: {
        fontWeight: 'bold',
    },
    oldPrice: {
        marginLeft: BaseStyles.margin.xs,
        fontWeight: 'bold',
        textDecorationLine: 'line-through',
    },
    discount: {
        marginLeft: BaseStyles.margin.xs,
        color: BaseStyles.colors.blue,
        fontWeight: 'bold',
    },
});

export default ProductStyles;
