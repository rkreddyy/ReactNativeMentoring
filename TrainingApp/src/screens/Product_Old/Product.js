import React, { useState } from "react";
import { ScrollView } from "react-native";
import Text from "../../components/Text";
import ProductImage from "../../components/ProductImage";
import Button from "../../components/Button";
import style from "./style";

const Product = React.memo(props => {
  const [loading, setLoading] = useState(false);
  const { product } = props.route.params;
  const { navigation } = props;
  navigation.setOptions({
    title: product.name
  });

  const onPress = async () => {
    setLoading(true);

    // Wait 5 seconds
    await new Promise(res => {
      setTimeout(res, 5000);
    });

    setLoading(false);
    navigation.goBack();
  };

  return (
    <React.Fragment>
      <ScrollView
        style={style.container}
        contentContainerStyle={style.contentContainer}
      >
        <ProductImage product={product} />
        <Text.TITLE style={style.title}>{product.name}</Text.TITLE>
        <Text>{product.description}</Text>
      </ScrollView>
      <Button loading={loading} onPress={onPress} title="Add to cart" />
    </React.Fragment>
  );
});

export default Product;
