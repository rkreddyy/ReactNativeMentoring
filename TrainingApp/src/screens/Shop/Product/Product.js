import React, {Component} from 'react';
import {View, Image, Text} from 'react-native';

import Styles from './styles';

export default class Product extends Component {
    getPriceString(item) {
        return `$${item.price}`;
    }

    getOldPriceString(item) {
        return `$${item.oldPrice}`;
    }

    getDiscountString(item) {
        return `${item.discount}% Off`;
    }

    render() {
        return (
            <View style={Styles.wrapper}>
                <View style={Styles.imageWrapper}>
                    <Image source={this.props.item.source} style={Styles.image} />
                </View>
                <Text>{this.props.item.name}</Text>
                <View style={Styles.priceWrapper}>
                    <Text style={Styles.price}>{this.getPriceString(this.props.item)}</Text>
                    <Text style={Styles.oldPrice}>{this.getOldPriceString(this.props.item)}</Text>
                    <Text style={Styles.discount}>{this.getDiscountString(this.props.item)}</Text>
                </View>
            </View>
        );
    }
}
