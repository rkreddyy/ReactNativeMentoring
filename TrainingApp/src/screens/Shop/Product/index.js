import Component from './Product'

export default Component
