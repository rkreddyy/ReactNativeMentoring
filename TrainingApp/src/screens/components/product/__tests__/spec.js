import React from 'react';
import { shallow } from 'enzyme';

import Product from '../product';

describe(`Product component`, () => {
    const renderComponent = props => shallow(<Product {...props} />);

    it('renders correctly', () => {
        const component = renderComponent({
            isExtended: true,
            item: { cell: { thumb: '' } },
            gotoProductDetails: () => { }
        });

        expect(component).toMatchSnapshot();
    });
});
