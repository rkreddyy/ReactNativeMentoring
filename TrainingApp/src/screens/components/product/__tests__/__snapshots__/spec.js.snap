// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "justifyContent": "center",
        "marginTop": 10,
        "overflow": "hidden",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http:",
        }
      }
      style={
        Object {
          "height": 250,
          "width": 250,
        }
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00a8f3",
          "justifyContent": "center",
          "marginLeft": "auto",
          "padding": 2,
          "width": 70,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        In Stock
      </Text>
    </View>
  </TouchableOpacity>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      $undefined
    </Text>
  </View>
</TouchableOpacity>
`;
